bootstrap/terraform/vm.tf

```diff
@@ -0,0 +1,218 @@
+  role_definition_name = "Reader"
+  scope                = azurerm_linux_virtual_machine.vault_azure_vm.id
+  principal_id         = azurerm_linux_virtual_machine.vault_azure_vm.identity[0].principal_id
+}
+
+# Uncomment below for user assigned managed identity
+resource "azurerm_user_assigned_identity" "vault_azure_uid" {
+  name                = "vault_azure_uid"
+  resource_group_name = azurerm_resource_group.vault_azure_rg.name
+  location            = azurerm_resource_group.vault_azure_rg.location
+}
+
+resource "azurerm_role_assignment" "app_assignment_vm_read" {
+  role_definition_name = "Reader"
+  scope                = azurerm_linux_virtual_machine.vault_azure_vm.id
+  principal_id         = azuread_service_principal.vault_azure_sp.object_id
+}
+
+resource "azurerm_resource_group" "vault_azure_rg" {
+  name     = "vault_azure_rg"
+  location = "westus2"
+}
+
+resource "azurerm_virtual_network" "vault_azure_vnet" {
+  name                = "vault_azure_vnet"
+  address_space       = ["10.0.0.0/16"]
+  location            = azurerm_resource_group.vault_azure_rg.location
+  resource_group_name = azurerm_resource_group.vault_azure_rg.name
+}
+
+resource "azurerm_subnet" "vault_azure_subnet" {
+  name                 = "internal"
+  resource_group_name  = azurerm_resource_group.vault_azure_rg.name
+  virtual_network_name = azurerm_virtual_network.vault_azure_vnet.name
+  address_prefixes     = ["10.0.2.0/24"]
+}
+
+resource "azurerm_network_interface" "vault_azure_nic" {
+  name                = "vault_azure_nic"
+  location            = azurerm_resource_group.vault_azure_rg.location
+  resource_group_name = azurerm_resource_group.vault_azure_rg.name
+
+  ip_configuration {
+    name                          = "internal"
+    subnet_id                     = azurerm_subnet.vault_azure_subnet.id
+    private_ip_address_allocation = "Dynamic"
+    public_ip_address_id          = azurerm_public_ip.vault_azure_pub_ip.id
+  }
+}
+
+resource "azurerm_public_ip" "vault_azure_pub_ip" {
+  name                = "vault_azure_pub_ip"
+  resource_group_name = azurerm_resource_group.vault_azure_rg.name
+  location            = azurerm_resource_group.vault_azure_rg.location
+  allocation_method   = "Static"
+}
+
+# Restrict SSH access to the local public IP
+data "http" "my_ip" {
+  url = "https://ifconfig.me/ip"
+}
+
+resource "azurerm_network_security_group" "vault_azure_sg" {
+  name                = "vault_azure_sg"
+  location            = azurerm_resource_group.vault_azure_rg.location
+  resource_group_name = azurerm_resource_group.vault_azure_rg.name
+
+  security_rule {
+    name                       = "ssh"
+    priority                   = 100
+    direction                  = "Inbound"
+    access                     = "Allow"
+    protocol                   = "Tcp"
+    source_port_range          = "*"
+    destination_port_range     = "22"
+    source_address_prefix      = chomp(data.http.my_ip.body)
```

Comment:
Getting a deprecation notice from terraform when running `make setup-env` for the body attribute. Maybe we should use [`response_body`](https://registry.terraform.io/providers/hashicorp/http/latest/docs/data-sources/http#body) instead?

Reply:
Good find. Done in [56208c3](https://github.com/hashicorp/vault-plugin-auth-azure/pull/65/commits/56208c3569d81279d1c75247f8ed8d4c118741f0).